encord/orm/dataset.py

```diff
@@ -54,19 +58,93 @@ class DatasetUsers:
     pass
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
 class DataClientMetadata:
     payload: dict
 
 
+@dataclasses.dataclass(frozen=True)
+class SignedUrl:
+    signed_url: str
+
+
+@dataclasses.dataclass(frozen=True)
+class ImageData:
```

Comment:
Note to self: This might need to be a class like the `DataRow` given that we might want to get signed urls individually in the case of very large image groups.